### PR TITLE
Add must rule for api audience (#348)

### LINKED ISSUE #348: Add must rule to mark APIs with intended audience
We propose a **must** rule to mark APIs with an intended API audience to support differentiated standards and richer infrastructure support for API discoverability, design and documentation quality, and permission granting (see [API Audience narrative](https://docs.google.com/document/d/1ff9b6oIa6dyQRyaj36-jmFdtCjWlngnGMsligR4RMIY) for more information).

### PR DESCRIPTION
**Pull request by tkrop**
Fixes #348

### CODE REVIEW

File: chapters/meta-information.adoc [hunk 1]
```diff
@@ -48,17 +52,87 @@ Example:
...
 ----
-"swagger": "2.0",
-"info": {
-  "title": "Parcel service API",
-  "description": "API for <...>",
-  "version": "1.3.7",
-    <...>
-}
+swagger: '2.0'
+info: {
```
Comment (maxim-tschumak) [inline chapters/meta-information.adoc]: `info: {` -> `info:`

File: chapters/meta-information.adoc [hunk 2]
```diff
@@ -48,17 +52,87 @@ Example:
...
+  product portfolio owned by the same business unit.
+*company-internal*::
+  The API consumers in this audience include all applications owned by the
+  business units of the same the company (e.g. Zalando company with Zalando
+  SE, Zalando Payments SE & Co. KG. etc.)
+*external-partner*::
+  The API consumers in this audience include all applications of business
+  partners of the company owning the API.
+*external-public*::
+  APIs in this audience can be accessed by anyone with Internet access.
```
Comment (maxim-tschumak) [inline chapters/meta-information.adoc]: Maybe `APIs in this audience` -> `APIs with this audience`?

File: chapters/meta-information.adoc (hunk 1 above)
Comment (maxim-tschumak) [reply chapters/meta-information.adoc]: `[source,json]` -> `[source,yaml]`

File: chapters/meta-information.adoc [hunk 3]
```diff
@@ -48,17 +52,87 @@ Example:
...
+----
+
+*Note:* Exactly *one audience* per API specification is allowed. (1) The
+smaller audience group is intentionally included in the wider group and thus
+does not need to be declared additionally. (2) We promote to split API
+specifications along the target audience — even if this creates redundancies
+(https://docs.google.com/document/d/1ff9b6oIa6dyQRyaj36-jmFdtCjWlngnGMsligR4RMIY/edit#heading=h.401bmun50mlj[rational]).
+
+For details and more information on audience groups see the
+https://docs.google.com/document/d/1ff9b6oIa6dyQRyaj36-jmFdtCjWlngnGMsligR4RMIY[
```
Comment (maxim-tschumak) [inline chapters/meta-information.adoc]: see the **internal** API Audience narrative

File: chapters/api-operation.adoc [hunk 4]
```diff
@@ -19,7 +19,8 @@ owned by a specific team and often implemented via small set of services. An
 external API is used by clients outside the team - usually another application
 owned by a different team or even an external business partner. An internal
 API is only used within the application and only by the owning team, for
-instance, for operational or implementation internal purposes.
+instance, for operational or implementation internal purposes. Internal APIs
+are <<219,marked with the target audience>> *component-internal*.
```
Comment (tfrauenstein) [inline chapters/api-operation.adoc]: This paragraph now is redundant with API Audience Rule. Let us simply delete it!

And add links to definition of internal vs. external (API Audience) in the 1st sentence of the rule.

File: chapters/meta-information.adoc (hunk 3 above)
Comment (tkrop) [reply chapters/meta-information.adoc]: fix on next line.

File: chapters/meta-information.adoc [hunk 5]
```diff
@@ -46,19 +50,89 @@ are free to decide whether they increment it or not.
...
+Each API must be classified with respect to the intended target *audience*
+supposed to consume the API, to facilitate differentiated standards on APIs
+for discoverability, changeability, quality of design and documentation, as
+well as permission granting. We differentiate the following API audience
+groups with clear organisational and legal boundaries: 
+
+*component-internal*::
+  The API consumers with this audience are restricted to applications of the
+  same https://docs.google.com/document/d/1ZSfVkdX_Dwpz22Xl-CFXgxe1u1eY_IfTNdFNMmnGi8c[
+  functional component [internal link]]
```
Comment (ePaul) [inline chapters/meta-information.adoc]: Missing `.` at the end.

File: chapters/meta-information.adoc [hunk 6]
```diff
@@ -48,17 +52,87 @@ Example:
...
+supposed to consume the API, to facilitate differentiated standards on APIs
+for discoverability, changeability, quality of design and documentation, as
+well as permission granting. We differentiate the following API audience
+groups with clear organisational and legal boundaries: 
+
+*component-internal*::
+  The API consumers in this audience are restricted to applications of the
+  same https://docs.google.com/document/d/1ZSfVkdX_Dwpz22Xl-CFXgxe1u1eY_IfTNdFNMmnGi8c[
+  functional component [internal link]]
+  All applications of a functional component are owned by specific dedicated
```
Comment (tfrauenstein) [inline chapters/meta-information.adoc]: editorial: applications --> service applications

File: chapters/meta-information.adoc [hunk 7]
```diff
@@ -46,19 +50,89 @@ are free to decide whether they increment it or not.
...
+*component-internal*::
+  The API consumers with this audience are restricted to applications of the
+  same https://docs.google.com/document/d/1ZSfVkdX_Dwpz22Xl-CFXgxe1u1eY_IfTNdFNMmnGi8c[
+  functional component [internal link]]
+  All applications of a functional component are owned by specific dedicated
+  owner and engineering team. Typical examples are APIs being used by internal
+  helper and worker services or that support service operation.
+*business-unit-internal*::
+  The API consumers with this audience include all applications of a specific
+  product portfolio owned by the same business unit.
```
Comment (ePaul) [inline chapters/meta-information.adoc]: I think the "includes" relation is the wrong way around here (and in the following items).

> Use this audience if all intended API consumers are applications of a ...

File: chapters/meta-information.adoc [hunk 8]
```diff
@@ -46,17 +50,87 @@ are free to decide whether they increment it or not.
...
+*company-internal*::
+  The API consumers with this audience include all applications owned by the
+  business units of the same the company (e.g. Zalando company with Zalando
+  SE, Zalando Payments SE & Co. KG. etc.)
+*external-partner*::
+  The API consumers with this audience include all applications of business
+  partners of the company owning the API.
```
Comment (ePaul) [inline chapters/meta-information.adoc]: "... and the company itself"?

File: chapters/meta-information.adoc [hunk 9]
```diff
@@ -46,17 +50,87 @@ are free to decide whether they increment it or not.
...
+  description: API for <...>
+  version: 1.2.4
+  <...>
+----
+
+*Note:* Exactly *one audience* per API specification is allowed. (1) The
+smaller audience group is intentionally included in the wider group and thus
+does not need to be declared additionally. (2) We promote to split API
+specifications along the target audience — even if this creates redundancies
+(https://docs.google.com/document/d/1ff9b6oIa6dyQRyaj36-jmFdtCjWlngnGMsligR4RMIY/edit#heading=h.401bmun50mlj[rational]).
```
Comment (ePaul) [inline chapters/meta-information.adoc]: "rational" → "rationale"? Also, this is an internal link (same document as the one below, but pointing to a section).

File: chapters/api-operation.adoc (hunk 4 above)
Comment (tkrop) [reply chapters/api-operation.adoc]: Should we delete the whole paragraph?

File: chapters/meta-information.adoc (hunk 9 above)
Comment (tkrop) [reply chapters/meta-information.adoc]: Moved after specification, in front of the example.

**Comment by ePaul** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by tfrauenstein** [comment]
👍

**Pull request merged**